cmd/restic/cmd_rewrite_integration_test.go

```diff
@@ -170,5 +171,133 @@ func TestRewriteSnaphotSummary(t *testing.T) {
 	rtest.OK(t, err)
 	rtest.Assert(t, sn.Summary != nil, "snapshot should have summary attached")
 	rtest.Equals(t, oldSummary.TotalBytesProcessed, sn.Summary.TotalBytesProcessed, "unexpected TotalBytesProcessed value")
-	rtest.Equals(t, oldSummary.TotalFilesProcessed, sn.Summary.TotalFilesProcessed, "unexpected TotalFilesProcessed value")
+}
+
+func TestRewriteInclude(t *testing.T) {
+	env, cleanup := withTestEnvironment(t)
+	defer cleanup()
+	createBasicRewriteRepo(t, env)
+	snapshots := testListSnapshots(t, env.gopts, 1)
+
+	// restic rewrite <snapshots[0]> -i "*.txt" --forget
+	rtest.OK(t, runRewrite(context.TODO(), RewriteOptions{
+		Forget:                true,
+		IncludePatternOptions: filter.IncludePatternOptions{Includes: []string{"*.txt"}}},
+		env.gopts,
+		[]string{snapshots[0].String()}))
+	newSnapshots := testListSnapshots(t, env.gopts, 1)
+	rtest.Assert(t, snapshots[0] != newSnapshots[0], "snapshot id should have changed")
+
+	// read restic ls output and count .txt files
+	out := testRunLsWithOpts(t, env.gopts, LsOptions{}, []string{newSnapshots[0].Str()})
+	fileList := strings.Split(string(out), "\n")
+	count := 0
+	for _, filename := range fileList {
+		if strings.Contains(filename, ".txt") {
+			count++
+		}
+	}
+	rtest.Assert(t, count == 2, "there should be 2 txt files in the snapshot, but there are %d files", count)
```

Comment:
Please extract this into a helper function. And also use it for `TestRewriteEmptyDirectory`.